backend/module/authn/authn.go

```diff
@@ -104,3 +109,32 @@ func (a *api) Callback(ctx context.Context, request *authnv1.CallbackRequest) (*
 		AccessToken: token.AccessToken,
 	}, nil
 }
+
+func (a *api) CreateToken(ctx context.Context, request *authnv1.CreateTokenRequest) (*authnv1.CreateTokenResponse, error) {
+	var prefixedSubject string
+	switch request.TokenType {
+	case authnv1.CreateTokenRequest_SERVICE:
+		prefixedSubject = "service:" + request.Subject
+	default:
+		return nil, status.Error(codes.InvalidArgument, "invalid token type")
+	}
+
+	var expiry *time.Duration
+
+	if request.Expiry != nil {
+		convertedExpiry, err := ptypes.Duration(request.Expiry)
```

Comment:
ptypes.Duration is deprecated, use durationpb package

this is enforced on latest main